test/Conversion/tosa-to-emitc.mlir

```diff
@@ -30,7 +30,63 @@ func @test_rsqrt(%arg0: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
+  // CHECK: emitc.call "tosa::mul"(%arg0, %arg1) {args = [1 : i32], template_args = []} : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  %0 = "tosa.mul"(%arg0, %arg1)  { shift = 1 : i32 } : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  return %0 : tensor<13x21x3xf32>
+}
+
+// MulOp: First operand needs to be broadcasted
+func @test_mul1(%arg0: tensor<13x1x3xf32>, %arg1: tensor<13x21x3xf32>) -> tensor<13x21x3xf32> {
+  // emitc.call "emitc::broadcast_in_dim"(%arg0) {args = [dense<[0, 1, 2]> : tensor<3xi64>], template_args = [tensor<13x21x3xf32>]} : (tensor<13x1x3xf32>) -> tensor<13x21x3xf32>
+  // CHECK: emitc.call "tosa::mul"(%0, %arg1) {args = [1 : i32], template_args = []} : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  %0 = "tosa.mul"(%arg0, %arg1)  { shift = 1 : i32 } : (tensor<13x1x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  return %0 : tensor<13x21x3xf32>
+}
+
+// MulOp: Second operand needs to be broadcasted
+func @test_mul2(%arg0: tensor<13x21x3xf32>, %arg1: tensor<13x1x3xf32>) -> tensor<13x21x3xf32> {
+  // CHECK: emitc.call "emitc::broadcast_in_dim"(%arg1) {args = [dense<[0, 1, 2]> : tensor<3xi64>], template_args = [tensor<13x21x3xf32>]} : (tensor<13x1x3xf32>) -> tensor<13x21x3xf32>
+  // CHECK: emitc.call "tosa::mul"(%arg0, %0) {args = [1 : i32], template_args = []} : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  %0 = "tosa.mul"(%arg0, %arg1)  { shift = 1 : i32 } : (tensor<13x21x3xf32>, tensor<13x1x3xf32>) -> tensor<13x21x3xf32>
+  return %0 : tensor<13x21x3xf32>
+}
+
+// MulOp: Second operand needs to be broadcasted + expanded
+func @test_mul3(%arg0: tensor<21x3xf32>, %arg1: tensor<3xf32>) -> tensor<21x3xf32> {
+  // CHECK: emitc.call "emitc::broadcast_in_dim"(%arg1) {args = [dense<1> : tensor<1xi64>], template_args = [tensor<21x3xf32>]} : (tensor<3xf32>) -> tensor<21x3xf32>
+  // CHECK: emitc.call "tosa::mul"(%arg0, %0) {args = [3 : i32], template_args = []} : (tensor<21x3xf32>, tensor<21x3xf32>) -> tensor<21x3xf32>
+  %0 = "tosa.mul"(%arg0, %arg1)  { shift = 3 : i32 } : (tensor<21x3xf32>, tensor<3xf32>) -> tensor<21x3xf32>
+  return %0 : tensor<21x3xf32>
+}
+
+// MulOp: Second operand needs to be broadcasted + expanded
+func @test_mul4(%arg0: tensor<13x21x3xf32>, %arg1: tensor<3xf32>) -> tensor<13x21x3xf32> {
+  // CHECK: emitc.call "emitc::broadcast_in_dim"(%arg1) {args = [dense<2> : tensor<1xi64>], template_args = [tensor<13x21x3xf32>]} : (tensor<3xf32>) -> tensor<13x21x3xf32>
+  // CHECK: emitc.call "tosa::mul"(%arg0, %0) {args = [1 : i32], template_args = []} : (tensor<13x21x3xf32>, tensor<13x21x3xf32>) -> tensor<13x21x3xf32>
+  %0 = "tosa.mul"(%arg0, %arg1)  { shift = 1 : i32 } : (tensor<13x21x3xf32>, tensor<3xf32>) -> tensor<13x21x3xf32>
+  return %0 : tensor<13x21x3xf32>
+}
+
+// MulOp: Second operand needs to be broadcasted + expanded with two dimension
```

Comment:
```suggestions
// MulOp: Second operand needs to be broadcasted + expanded by two dimensions
```
By two dimensions or by a second? Haven't looked that closely.

Reply:
I updated the comment in the test to make it more clear:

`// MulOp: Second two dimensional operand needs to be broadcasted + expanded to four dimensions`

A broadcast from a Tensor2D into a Tensor4D is happening here.